src/thor/map_matcher.cc

```diff
@@ -153,39 +173,43 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
       // This means there is a discontinuity. If so, fallback to using costing
       use_timestamps = false;
     }
-    last_interp_index = interpolations.front().back().original_index;
+    if (!interpolations.empty())
+      last_interp_index = interpolations.front().back().original_index;
```

Comment:
@kdiluca i dont remember making this change can you speak to it? i can see its to protect against dereferencing something that doesnt exist but it defaults to an index of 0 so maybe its not all the protection we need. have to look below i guess.

Reply:
oh i see, we disable use of interpolated timestamps entirely if there arent any. why not just move this inside the if above

Reply:
ok ive rejiggered thsi to avoid any possible access of interpolations that doesnt make sense